src/test/TestUtils/TestProjectFixture.cs

```diff
@@ -310,15 +317,10 @@ public TestProjectFixture EnsureRestored(params string[] fallbackSources)
 
         public TestProjectFixture EnsureRestoredForRid(string rid, params string[] fallbackSources)
         {
-            var sourceProjectJson = Path.Combine(_testProject.ProjectDirectory, "project.json.template");
-            var targetProjectJson = Path.Combine(_testProject.ProjectDirectory, "project.json");
-
-            // apply RID to template
-            File.WriteAllText(targetProjectJson, File.ReadAllText(sourceProjectJson).Replace("{RID}", rid));
-
             if ( ! _testProject.IsRestored())
             {
-                RestoreProject(fallbackSources);
+                string extraMSBuildProperties = $"/p:TestTargetRid={rid}";
```

Comment:
As above, I have a preference for this method. Most of the test projects do not define `RuntimeIdentifiers` or `RuntimeIdentifier`. Those that do can define it to be based on this auxiliary property.